File: public/docs/_examples/dependency-injection/ts/app/heroes/hero.service.2.ts
// #docregion
import { Injectable } from '@angular/core';

import { Hero }       from './hero';
import { HEROES }     from './mock-heroes';
import { Logger }     from '../logger.service';

@Injectable()
export class HeroService {

  //#docregion ctor
  constructor(private logger: Logger) {  }
  //#enddocregion ctor

  getHeroes() {
    this.logger.log('Getting heroes ...')
    return HEROES;
  }
}

File: public/docs/_examples/dependency-injection/dart/lib/heroes/hero_service_2.dart
// #docregion
import 'package:angular2/core.dart';

import '../logger_service.dart';
import 'hero.dart';
import 'mock_heroes.dart';

@Injectable()
class HeroService {
  final Logger _logger;

  //#docregion ctor
  HeroService(this._logger);
  //#enddocregion ctor
  List<Hero> getHeroes() {
    _logger.log('Getting heroes ...');
    return HEROES;
  }
}
